### PR TITLE
Automatically generate a calibration matrix for libinput

### PR DESCRIPTION
**Pull request by michitux**
This takes the calibration matrix for Wayland and divides the pixel offsets by the device width/height. This is necessary as libinput for X.org seems to use coordinates between 0 and 1.

Please discuss if this is the right approach, if there should be more precautions/checks and what kind of tests would be necessary. I have only tested this for my own device (Samsung Galaxy Note 8.0). The configuration for the Nokia N900 is another example for a device with a non-trivial calibration matrix that is also adjusted for libinput for X.org.

### CODE REVIEW

File: aports/main/devicepkg-dev/devicepkg_build.sh [hunk 1]
```diff
@@ -20,6 +20,22 @@ fi
...
+		# shellcheck disable=SC2154
+		x_offset=$(dc "$3" "$deviceinfo_screen_width" / p)
+		# shellcheck disable=SC2154
+		y_offset=$(dc "$6" "$deviceinfo_screen_height" / p)
+		if [ ! -z "$x_offset" ] && [ ! -z "$y_offset" ]
+		then
+			echo "ENV{LIBINPUT_CALIBRATION_MATRIX}=\"$1 $2 $x_offset $4 $5 $y_offset\", \\"
+		fi
+	fi
+}
```
Comment (ollieparanoid) [inline aports/main/devicepkg-dev/devicepkg_build.sh]: This function does some checking before writing the line to stdout, which is good!
How about writing to `stderr` when a condition does not match, to make debugging easier?

File: aports/main/devicepkg-dev/APKBUILD [hunk 2]
```diff
@@ -16,5 +16,5 @@ package() {
 	install -Dm755 "$srcdir/devicepkg_package.sh" \
 		"$pkgdir/usr/bin/devicepkg_package"
 }
-sha512sums="e6f96f8942fa8124ca7461bcf9c78879e08ef00da4a728735046503cae86548505594e078eb14056af74616b17332474f1473f72127299773e075bfd970af13b  devicepkg_build.sh
+sha512sums="9316aad68cf4db3f17cf11511673e16c9cbc02d75cf7963ba2f1daf6cfa6fbd9342b1d739b73e745fb076250db0a3921ae4eedec58efc70acc06a1c6614039c2  devicepkg_build.sh
```
[the remark this answers is not in the document]
Comment (michitux) [reply aports/main/devicepkg-dev/APKBUILD]: Sorry for this. I have updated the checksum, the build is passing now.

File: aports/main/devicepkg-dev/devicepkg_build.sh [hunk 3]
```diff
@@ -20,6 +20,22 @@ fi
 # shellcheck disable=SC1090,SC1091
 . "$srcdir/deviceinfo"
 
+echo_libinput_calibration()
```
[the remark this answers is not in the document]
Comment (michitux) [reply aports/main/devicepkg-dev/devicepkg_build.sh]: I fear there is no really short description but I have added a description for each parameter now.

File: aports/main/devicepkg-dev/devicepkg_build.sh [hunk 4]
```diff
@@ -20,6 +20,22 @@ fi
...
+{
+	# shellcheck disable=SC2154
+	if [ $# -eq 6 ] && [ ! -z "$deviceinfo_screen_width" ] && [ ! -z "$deviceinfo_screen_height" ]
```
[the remark this answers is not in the document]
Comment (michitux) [reply aports/main/devicepkg-dev/devicepkg_build.sh]: Okay, makes sense, I have changed that.

File: aports/main/devicepkg-dev/devicepkg_build.sh (hunk 1 above)
Comment (michitux) [reply aports/main/devicepkg-dev/devicepkg_build.sh]: Good point, I have restructured the code to print a message whenever a condition fails.

**Comment by ollieparanoid** [comment]
Thanks again! I've taken the liberty to break the long error messages into two lines. We don't have strict guidelines for shell script formatting yet (the idea of having Travis enforce that is in #110). But since they were pretty long it made sense to shorten them.
(If you would have preferred that I let you confirm this change before merging it, just tell me and we'll do it that way with your future PRs.)

I have tested the modified `devicepkg-dev` script. The udev file gets generated as expected, and the error messages work as well.

**Pull request merged**